src/activatable/__tests__/activatableCollection.unit.ts

```diff
@@ -0,0 +1,34 @@
+import { Activatable, ActivatableCollection } from '../';
+
+const sleep = (ms: number) => new Promise((resolve, reject) => setTimeout(resolve, ms));
+
+describe('activatable', () => {
+  test('collection', async () => {
```

Comment:
Let's add an additional test case for the "one activable fails to activate" scenario.

Reply:
what's going on in the example?

Reply:
> (are you trying to test some kind of race condition?)

Yeah, I know we are not prone to this in any case now, so well, we can skip that one.
Basically, given something as follows:
```ts
  test('foo', async done => {
    class MyActivatableThing2 extends Activatable {
      protected async doActivate() {
        await sleep(100);
      }
      protected async doDeactivate() {
        await sleep(300);
      }
    }

    const activatables = new ActivatableCollection();
    activatables.push(new MyActivatableThing());
    activatables.push(new MyActivatableThing2());
    await activatables.activate();
    await activatables.deactivate();
    setTimeout(() => {
      expect(activatables.state).toEqual('activated');
      done();
    }, 301); // suggest using fake timers
  });
```
and a broken code (note, typings do not enforce promise to be returned, which is wrong IMHO)
```ts
  protected  doActivate() {
    for (const activatable of this.activatables) {
       activatable.activate();
    }
  }

  protected async doDeactivate() {
    for (const activatable of this.activatables) {
       activatable.deactivate();
    }
  }
```
The test case would fail.
Not a big deal, though.

Reply:
Ah. Still not sure what the setTimeout is for, since you're awaiting `activate` and `deactivate`.

I agree about the types though.

Reply:
Timeout is not needed, I should have written that in webstorm, not this textarea thingy lol.
Either way, you get the point now.